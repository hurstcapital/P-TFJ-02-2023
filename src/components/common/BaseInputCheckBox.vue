<template>
  <input
    data-cy="check-box"
    @input="updateInput"
    type="checkbox"
    :checked="isChecked"
    class="accent-green-500"
  />
</template>

<script lang="ts">
import { defineComponent } from "vue";

export default defineComponent({
  name: "BaseInputCheckBox",
  props: {
    isChecked: {
      type: Boolean,
      default: false,
    },
  },
  setup(_, ctx) {
    const updateInput = (event: Event) => {
      ctx.emit("update:checked", (event.target as HTMLInputElement).value);
    };
    return { updateInput };
  },
});
</script>
