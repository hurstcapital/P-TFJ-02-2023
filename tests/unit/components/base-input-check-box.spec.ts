import { shallowMount } from "@vue/test-utils";
import BaseInputCheckBox from "@/components/common/BaseInputCheckBox.vue";

describe("BaseInputCheckBox.vue", () => {
  it("should renders checked", () => {
    const wrapper = shallowMount(BaseInputCheckBox, {
      props: {
        checked: false,
      },
    });

    expect(wrapper.element).toMatchSnapshot();
  });
});
