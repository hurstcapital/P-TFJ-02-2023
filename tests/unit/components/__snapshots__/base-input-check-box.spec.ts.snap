// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseInputCheckBox.vue should renders checked 1`] = `
<input
  class="accent-green-500"
  data-cy="check-box"
  type="checkbox"
/>
`;
